### PR TITLE
[Debug UI] Dockerize debug UI project for deployment.

### PR DESCRIPTION
**Pull request by robin-near**
This is so we can easily deploy it to a server we bring up.

#8355

### CODE REVIEW

File: tools/debug-ui/src/index.tsx [hunk 1]
```diff
@@ -16,7 +17,7 @@ const router = createBrowserRouter([
         // Root path redirects to the current host - this is useful if we serve the app from the node
         // itself.
```
Comment (soonnear) [inline tools/debug-ui/src/index.tsx:18]: Are these comments still up to date?

File: tools/debug-ui/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,15 @@
+FROM node:19-alpine AS build
```
Comment (soonnear) [inline tools/debug-ui/Dockerfile:1]: Not everyone might be familiar with Docker and/or nginx. 
Could you add code script instructions into the README.md on how to use this and run this locally?
Thanks!

File: tools/debug-ui/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+RUN npm install
+
+ADD src /build/src
+ADD public /build/public
+ADD tsconfig.json /build/tsconfig.json
+RUN npm run build
+
+FROM nginx:1.21-alpine
+COPY --from=build /build/build /var/www/html
+COPY nginx.conf /etc/nginx/conf.d/default.conf
```
Comment (soonnear) [inline tools/debug-ui/Dockerfile:15]: Why isn't npm start (starting the local server) part of the docker image? 
Are we supposed to run this container then start it with a separate command?

**Comment by akhi3030** [comment]
@soonnear, @robin-near: let me know when this is ready and I'll approve it.

**Review by soonnear — APPROVED** [review]
LGTM

**Review by akhi3030 — APPROVED** [review]

File: tools/debug-ui/src/index.tsx [hunk 4]
```diff
@@ -2,10 +2,11 @@ import * as React from 'react';
 import * as ReactDOM from 'react-dom/client';
 import './index.css';
 import '@patternfly/react-core/dist/styles/base.css';
-import { createBrowserRouter, Navigate, RouterProvider } from 'react-router-dom';
+import { createBrowserRouter, RouterProvider } from 'react-router-dom';
 import { QueryClient, QueryClientProvider } from 'react-query';
 import { App } from './App';
 import { LogVisualizer } from './log_visualizer/LogVisualizer';
+import { LandingPage } from './LandingPage';
```
[the remark this answers is not in the document]
Comment (robin-near) [reply tools/debug-ui/src/index.tsx:9]: I'm struggling with making eslint do just enough things, not too much, not to little. It has a million configs :( Feel free to tweak it, but in the meantime I don't think the project is big enough that import order matters all that much.

[the remark this answers is not in the document]
Comment (soonnear) [reply tools/debug-ui/src/index.tsx:9]: No worries, this is just a nit comment. I will see if I can update eslint to fix the import statements.

File: tools/debug-ui/Dockerfile (hunk 3 above)
Comment (soonnear) [reply tools/debug-ui/Dockerfile:17]: Thanks! That makes sense 

Never used nginx before and thought it was used only as a backwards proxy (caches for instance) for handling many incoming request, so wasn't aware it's also a general purpose web server.

**Pull request merged**